--- a/src/common/net/net.cpp
+++ b/src/common/net/net.cpp
@@ -240,7 +240,7 @@ std::optional<std::string> MakeRequest(const std::string& url, const std::string
                       response.status);
             return {};
         }
-        if (!response.headers.contains("content-type")) {
+        if (response.headers.count("content-type") == 0) {
             LOG_ERROR(Common, "GET to {}{} returned no content", url, path);
             return {};
         }
--- a/src/yuzu/updater/update_dialog.cpp
+++ b/src/yuzu/updater/update_dialog.cpp
@@ -160,7 +160,7 @@ void UpdateDialog::Download() {
                                               QString::number(response.status)));
         return;
     }
-    if (!response.headers.contains("content-type")) {
+    if (response.headers.count("content-type") == 0) {
         LOG_ERROR(Frontend, "GET to {}{} returned no content", m_asset.url, m_asset.path);
         return;
     }
